### PR TITLE
use correct path when running php unit tests

### PR DESCRIPTION
**Pull request by senadir**
When running PHP unit tests, we mount the current folder to docker, to have flexibility, we assign the current folder name in docker instead of hardcoding.

The part that inferred the folder name didn't take into account that folders and parent folders might have spaces in them, switching from `$(basename $(pwd))` to `$(basename "$(pwd)"` because when passing `$(pwd)` directly with unescaped spaces, you're passing N values to basename, when wrapping it in "" you're passing a single value.

### Testing

- Change the parent folder name of your folder to something with spaces
- Run `npm run test:php`, it should work fine.
- Rename the current folder to something with spaces, run the command again, it should work.

### CODE REVIEW

**Review by dinhtungdu — APPROVED** [review]
LGTM! 🎉 

Screen Shot 2022-10-26 at 23 16 15

**Comment by danieldudzic** [comment]
Hmm, it still seems to be broken for me:

danieldudzic_Dannys-MacBook____Local_Sites_ratings_app_public_wp-content_plugins_woocommerce-blocks_and_use_correct_path_when_running_php_unit_tests_by_senadir_·_Pull_Request__7489_·_woocommerce_woocommerce-blocks

**Review by alexflorisca — APPROVED** [review]
This works for me too, but I wonder why it doesn't work for @danieldudzic? Maybe let's get to the bottom of that before we merge

**Comment by ralucaStan** [comment]
Why does this PR have the on hold label @senadir ?

**Comment by wavvves** [comment]
It does not work for me too:
image

@dinhtungdu I see the path for your PHPUnit conf does not have spaces, and I'm wondering if Alex's was the same. Can you re-test using a clean copy? I suspect wp-env is not updating paths.

EDIT: spaces could be in the upper directory, which should be trimmed. Anyway, please try a fresh clone into a dir with spaces and clean wp-env

In the meantime, I'll test some changes to try and fix this.

File: package.json
Comment (wavvves) [inline package.json]: Hey @nielslange previous fix does not work, and there's no need to use basename here. Still I feel you can never really win with these things, we just keep trading bugs 🙄😂

```suggestion
		"test:php": "npm run wp-env run phpunit '/var/www/html/wp-content/plugins/${PWD##*/}/vendor/bin/phpunit -c /var/www/html/wp-content/plugins/${PWD##*/}/phpunit.xml.dist'",
```

**Comment by nielslange** [comment]
> It does not work for me too

Thanks for reviewing this PR, @wavvves. Initially, I faced the same problem. I solved the problem by running the following command:

```sh
npm run wp-env destroy && npm run wp-env start
```

I hope that commend helps to run `npm run test:php` successfully.

**Comment by wavvves** [comment]
> > It does not work for me too
> 
> 
> 
> Thanks for reviewing this PR, @wavvves. Initially, I faced the same problem. I solved the problem by running the following command:
> 
> 
> 
> ```sh
> 
> npm run wp-env destroy && npm run wp-env start
> 
> ```
> 
> 
> 
> I hope that commend helps to run `npm run test:php` successfully. 

I cleared out wp-env and started fresh with a new git clone and problem still persisted.

**Review by wavvves — CHANGES_REQUESTED** [review]

File: package.json
Comment (nielslange) [reply package.json]: > Hey @nielslange previous fix does not work, and there's no need to use basename here. Still I feel you can never really win with these things, we just keep trading bugs 🙄😂

Thanks for the suggestions. Can you think of a better approach to solve the current problem?

Comment (wavvves) [reply package.json]: That is the best alternative I can think of 😅

Comment (nielslange) [reply package.json]: Given that this is the best alternative you can think of, I'm happy to accept your suggestion, @wavvves! 🙏

**Comment by dinhtungdu** [comment]
> @dinhtungdu I see the path for your PHPUnit conf does not have spaces, and I'm wondering if Alex's was the same. Can you re-test using a clean copy? I suspect wp-env is not updating paths.

@wavvves I tested using a clean copy but the initial fix still works for me.

I also tested your suggestion and it works for me too. I prefer yours because it is shorter and cleaner.

cc @nielslange

**Pull request merged**
